tests/test_charm.py

```diff
@@ -8,8 +8,73 @@
 
 
 class TestCharm(unittest.TestCase):
+    def setUp(self) -> None:
+        self.harness = Harness(MySQLCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+        self.harness.add_oci_resource("mysql-image")
 
-    def setUp(self):
-        harness = Harness(MySQLCharm)
-        self.addCleanup(harness.cleanup)
-        harness.begin()
+    def test_hostname_for_leader_unit(self):
+        self.harness.set_leader(True)
+        expected_host = "mysql-0.mysql-endpoints"
+        self.assertEqual(self.harness.charm.hostname, expected_host)
+
+    def test_env_config(self):
```

Comment:
Here is my humble opinion, take it for whatever it is worth. Seems to me you are testing implementation rather than functionality of your Charm in this and a few other unit tests. Your charm has not implemented the code that makes environment variables work. This code is part of the Operator Framework/Juju/Testing Harness. Hence I see no reason why your charm should be testing that environment variables work as expected. On the other hand if one specific environment variable changes the behaviour of your charm in a specific way, then yes a unit test to check that the behaviour of your charm changes in response to that environment variable will be useful.